lib/driver.js

```diff
@@ -256,15 +259,21 @@ class AndroidUiautomator2Driver extends BaseDriver {
       await this.initAUT();
     }
     //Adding AUT package name in the capabilities if package name not exist in caps
-    if (!this.caps.appPackage) {
-      this.caps.appPackage = appInfo.appPackage;
+    if (this.opts.appInfo && !this.caps.appPackage) {
```

Comment:
this is not going to work. no `appInfo` property will ever exist on `this.opts`. The line above (`Object.assign(this.opts, appInfo);`) does not create a property called `appInfo`. It puts all the properties that exist on the `appInfo` object into `this.opts`. If you read the implementation of `helpers.getLaunchInfo`, you'll see that the last line is:

```js
return {appPackage, appWaitPackage, appActivity, appWaitActivity};
```

which means these are the properties that get updated on `this.opts` as a result of the `Object.assign`.